python/cuml/manifold/umap.pyx

```diff
@@ -364,17 +272,6 @@ cdef class UMAP:
         del X_m
```

Comment:
Missing a call to `handle.sync()` to make sure the fit is complete before deleting `X_m1

Reply:
Looking into cython a little bit more, it does appear that have an API for supporting asynchronous functions [1] , but it needs to be explicitly used with `async def funcName()`. Otherwise, I would expect these C++ functions to block until complete. If they did not do this, I believe we would see other problems in the tests and notebooks. 

[1] https://www.python.org/dev/peps/pep-0492/